src/main/java/io/knowledgeassets/myskills/server/community/command/CommunityUserCommandController.java

```diff
@@ -71,11 +77,14 @@ public CommunityUserCommandController(CommunityUserCommandService communityUserC
 			@ApiResponse(code = 500, message = "Error during execution")
 	})
 	@DeleteMapping(path = "/communities/{communityId}/users/{userId}")
-	@PreAuthorize("isAuthenticated() and isPrincipalUserId(#userId)")
+	@PreAuthorize("(isAuthenticated() and isPrincipalUserId(#userId)) or hasCommunityManagerRole(#communityId)")
 	public ResponseEntity<CommunityResponse> leaveCommunity(@PathVariable("communityId") String communityId,
 															@PathVariable("userId") String userId) {
+		final Community community = communityUserCommandService.leaveCommunity(communityId, userId);
+		final CommunityResponse communityResponse = securityService.hasCommunityManagerRole(SecurityContextHolder.getContext().getAuthentication().getPrincipal(), communityId) ?
```

Comment:
Instead of calling the `SecurityContext` methods directly you can inject the authentication's principal as a method argument in the controller method by using `@AuthenticationPrincipal`. See: https://docs.spring.io/spring-security/site/docs/5.1.4.RELEASE/reference/htmlsingle/#mvc-authentication-principal

Reply:
Thanks, done.